docs/proposals/scheduling/20220701-schedule-gang.md

```diff
@@ -0,0 +1,421 @@
+---
+title: Gang scheduling
+authors:
+  - "@buptcozy"
+reviewers:
+  - "@eahydra"
+  - "@hormes"
+  - "@yihuifeng"
+  - "@honpey"
+  - "@zwzhang0107"
+  - "@jasonliu747"
+creation-date: 2022-07-01
+last-updated: 2022-07-01
+
+---
+
+# Gang scheduling
+
+## Table of Contents
+
+<!--ts-->
+
+* [Gang scheduling](#Gang-scheduling)
+    * [Table of Contents](#table-of-contents)
+    * [Summary](#summary)
+    * [Motivation](#motivation)
+      * [Compared with competitors](#Compared-with-competitors)
+        * [Coescheduling](#Coescheduling)
+      * [Goals](#goals)
+      * [Non Goals and Future Work](#Non-Goals-and-Future-Work)
+    * [Proposal](#Proposal)
+      * [Key concept](#key-concept)
+        * [Strict-mode and non-strict-mode](#strict-mode-and-non-strict-mode)
+        * [GangGroup](#ganggroup)
+        * [After gang](#after-gang)
+      * [API](#API)
+        * [Definition](#definition)
+          * [Annotation way](#annotation-way)
+            * [Example](#example)
+           * [CRD way](#crd-way)
+             * [Example](#example)           
+      * [Implementation Details](#Implementation-Details)
+        * [QueueSortPlugin](#QueueSortPlugin)
+        * [Data-Structure](#data-structure)
+        * [GangPlugin](#gang-plugin)
+    * [Unsolved Problems](#Unsolved-Problems)
+    * [Alternatives](#Alternatives)
+    * [Implementation History](#Implementation-History)
+    * [References](#References)
+<!--te-->
+
+## Summary
+This proposal provides Gang mechanism for the scheduler to control pods binding opportunity. User can declare a resource-collection-minimum number, 
+only when assigned-resources reach the given limitation can trigger the binding. We provide `strict-mode` and `non-strict-mode` to 
+control the resource-accumulation-process by a configuration. We also provide a two-level Gang description for better matching 
+the real scenario, which is different from community.
+
+## Motivation
+In AI scenarios, lots of jobs need Gang scheduling. The community have lots of related implements such as `coescheduling` or `vocalno`.
+However, in practice we find that the above solutions do not meet the needs of the business in some places.
```

Comment:
Don't diss other solutions, praise them and enhance it.

Reply:
fixed

Reply:
fixed